src/Form/atoms/ButtonWithLoading.tsx

```diff
@@ -54,26 +55,32 @@ const LoadingContainer = styled.div<{ design: TypeButtonDesigns, show?: boolean,
   }
 `;
 
-const InnerContainer = styled.div<{position?: string, loading: string, design: TypeButtonDesigns}>`
+const InnerContainer = styled.div<{position?: string, loading: string, design: TypeButtonDesigns, size: TypeButtonSizes}>`
   display: flex;
   height: inherit;
+  ${({size}) => (size === 'extraSmall') ? css`margin-top: -3px`: ``};
 
-
-  ${({ position, loading }) => position && position === 'left' ? css`
-    margin-right: ${ loading === 'true' ? '-20px' : '0' };
+  ${({ position, loading, size }) => position && position === 'left' ? css`
+      margin-right: ${ loading === 'true'
+      ?  (size === 'extraSmall') ? '-10px' : '-20px'
+      : '0' 
+    };
   ` : css`
-    margin-left: ${ loading === 'true' ? '-20px' : '0' };
+      margin-left: ${ loading === 'true'
+      ? (size === 'extraSmall') ? '-10px' : '-20px'
+      : '0'
+    };
   `}
```

Comment:
These negative value margin tweaks shouldn't be needed. I had a quick look and it seems the padding that has been added is why you're needing to compensate like this here for the small size buttons so I think once that padding has been addressed it should make a lot (if not all of this) redundant.